inductiva/simulators/simulator.py

```diff
@@ -54,6 +55,19 @@ def _setup_input_dir(self, input_dir: types.Path):
                 f"The provided path (\"{input_dir}\") is not a directory.")
         return input_dir
 
+    def _commands_to_json(self, kwargs):
+        if not "commands" in kwargs:
+            return kwargs
+
+        def to_cmd(x):
+            if isinstance(x, inductiva.commands.Command):
+                return x
+            return inductiva.commands.Command(x)
```

Comment:
Here, we need an `else` to capture the case when `x` is not a just a str. Moreover, I think we can serialize to json immediately and update the namings as follows:

```suggestion
        def to_cmd_json(cmd):
            if isinstance(cmd, inductiva.commands.Command):
                return cmd.to_json()
            elif not isinstance(cmd, str):
                raise ValueError("Command structure not valid. It should be of type: 'str' or 'inductiva.commands.Command'.")
            
            return inductiva.commands.Command(cmd).to_json()
```

Reply:
> Here, we need an `else` to capture the case when `x` is not a just a str. Moreover, I think we can serialize to json immediately and update the namings as follows:

But it should be either a `string` or a `Command` object. As long as that is clear to the user it should be fine, no?

Reply:
I will raise the error regardless

Reply:
You never know. They may pass a tuple to check if it is possible! And with an explicit error, they will know how to solve it without going to docs.